software/database/src/main/java/brooklyn/entity/database/mysql/MySqlSshDriver.java

```diff
@@ -186,9 +186,13 @@ protected void copyDatabaseConfigScript() {
     }
 
     protected boolean copyDatabaseCreationScript() {
-        InputStream creationScript = DatastoreMixins.getDatabaseCreationScript(entity);
-        if (creationScript==null) return false;
-        getMachine().copyTo(creationScript, getRunDir() + "/creation-script.sql");
+        String creationScriptContents = DatastoreMixins.getDatabaseCreationScriptAsString(entity);
+        if (creationScriptContents==null) return false;
+
+        creationScriptContents = processTemplateContents(creationScriptContents);
+        Reader creationContents = new StringReader(creationScriptContents);
+
+        getMachine().copyTo(creationContents, getRunDir() + "/creation-script.sql");
```

Comment:
Although `copyTemplate` wants the URI of the template so its a bit awkward with `CREATION_SCRIPT_CONTENTS`. I guess you'd have to write it out as a temporary file. I leave it up to you.